multiqc/modules/preseq/preseq.py

```diff
@@ -95,22 +190,45 @@ def preseq_length_trimmed_plot (self):
             'ymin': 0,
             'xmin': 0,
             'tt_label': '<b>{point.x:,.0f} total</b>: {point.y:,.0f} unique',
-            'extra_series': [{
-                'name': 'x = y',
-                'data': [[0, 0], [self.total_max, self.total_max]],
-                'dashStyle': 'Dash',
-                'lineWidth': 1,
-                'color': '#000000',
-                'marker': { 'enabled': False },
-                'enableMouseTracking': False,
-                'showInLegend': False,
-            }]
+            'extra_series': []
         }
         description = ''
-        if xmax is not None:
-            pconfig['xmax'] = xmax
-            description += "Note that the x axis is trimmed until one of the datasets \
-                shows 90% of its maximum y-value, to avoid ridiculous scales."
+        
+        # Plot the real counts if we have them
+        real_counts_total, real_counts_unique = self._read_real_counts()
+        pconfig['extra_series'].extend(self._real_counts_to_plot_series(real_counts_unique, real_counts_total))
+        if real_counts_unique:
+            description += '<p>Points show read count versus deduplicated read counts (externally calculated).</p>'
+        elif real_counts_total:
+            description += '<p>Points show externally calculated read counts on the curves.</p>'
+
+        # Trim the data to not have a ridiculous x-axis (10Gbp anyone?)
+        if getattr(config, 'preseq', {}).get('notrim', False) is not True:
+            ymax = min(max(d.values()) for d in self.preseq_data.values()) * 0.9
+            xmaxs = []
+            for sn, d in self.preseq_data.items():
+                xmax = 0
+                for x in sorted(list(d.keys())):
+                    xmax = max(xmax, x)
+                    if d[x] > ymax and x > real_counts_total.get(sn, 0) and x > real_counts_unique.get(sn, 0):
+                        break
+                xmaxs.append(xmax)
+            pconfig['xmax'] = min(xmaxs)
+            description += "<p>Note that the x axis is trimmed until one of the datasets \
+                shows 90% of its maximum y-value, to avoid ridiculous scales.</p>"
+        else:
+            ymax = max(max(d.values()) for d in self.preseq_data.values())
+            
+        pconfig['extra_series'].append({  # Plot perfect library as dashed line
```

Comment:
The dash line needs `ymax` to be defined first. I removed `self.total_max` and instead using `ymax` for plotting the dashed line. It's more accurate this way and also doesn't need to specifically convert `self.total_max` to coverage.